sdk/cosmos/azure-cosmos-spark_3_2-12/src/main/scala/com/azure/cosmos/spark/PartitionMetadataCache.scala

```diff
@@ -176,27 +243,70 @@ private object PartitionMetadataCache extends BasicLoggingTrait {
       .to(clientCacheItem => {
         val container = ThroughputControlHelper.getContainer(userConfig, cosmosContainerConfig, clientCacheItem.client)
 
-        val options = CosmosChangeFeedRequestOptions.createForProcessingFromNow(
+        val optionsFromNow = CosmosChangeFeedRequestOptions.createForProcessingFromNow(
           SparkBridgeImplementationInternal.toFeedRange(feedRange))
-        options.setMaxItemCount(1)
-        options.setMaxPrefetchPageCount(1)
-        options.setQuotaInfoEnabled(true)
+        optionsFromNow.setMaxItemCount(1)
+        optionsFromNow.setMaxPrefetchPageCount(1)
+        optionsFromNow.setQuotaInfoEnabled(true)
 
         val lastDocumentCount = new AtomicLong()
         val lastTotalDocumentSize = new AtomicLong()
         val lastContinuationToken = new AtomicReference[String]()
+        val firstLsn = new AtomicLong(-1)
 
-        container
-          .queryChangeFeed(options, classOf[ObjectNode])
-          .handle(r => {
+        val fromNowFlux: Flux[Int] = container
+          .queryChangeFeed(optionsFromNow, classOf[ObjectNode])
+          .byPage
+          .take(1)
+          .map(r => {
             lastDocumentCount.set(r.getDocumentCountUsage)
             lastTotalDocumentSize.set(r.getDocumentUsage)
             val continuation = r.getContinuationToken
+            logDebug(s"PartitionMetadataCache.readPartitionMetadata-fromNow(" +
+              s"${cosmosContainerConfig.database}.${cosmosContainerConfig.container}, $feedRange, $continuation)")
             if (!Strings.isNullOrWhiteSpace(continuation)) {
               lastContinuationToken.set(continuation)
             }
+            0
```

Comment:
WHy is this 0?

Reply:
Or is this a throwaway value and the only thing you do is capture the quota info and continuation?